addons/purchase_stock/tests/test_purchase_delete_order.py

```diff
@@ -40,3 +40,53 @@ def test_00_delete_order(self):
         purchase_order_3.button_cancel()
         self.assertEqual(purchase_order_3.state, 'cancel', 'PO is cancelled!')
         purchase_order_3.unlink()
+
+    def test_01_delete_propagation(self):
+        ''' Testcase for deleting purchase order with linked move and propagate cancel off'''
+
+        partner = self.env['res.partner'].create({'name': 'My Partner'})
+        product = self.env['product.product'].create({
+            'name': 'Test',
+            'type': 'product',
+        })
+
+        stock_location = self.env.ref('stock.warehouse0').out_type_id.default_location_src_id
+        cust_location = self.env.ref('stock.stock_location_customers')
+        picking_type_out = self.ref('stock.picking_type_out')
+        picking_out = self.env['stock.picking'].create({
+            'location_id': stock_location.id,
+            'location_dest_id': cust_location.id,
+            'partner_id': partner.id,
+            'picking_type_id': picking_type_out,
+        })
+        move = self.env['stock.move'].create({
+            'name': product.name,
+            'product_id': product.id,
+            'product_uom_qty': 1,
+            'product_uom': product.uom_id.id,
+            'picking_id': picking_type_out,
```

Comment:
Alternatively, what you you could do the following:
```suggestion
        move = self.env['stock.move'].create({
            'name': product.name,
            'product_id': product.id,
            'product_uom_qty': 1,
            'product_uom': product.uom_id.id,
            'picking_type_id': picking_type_out,
            'location_id': stock_location.id,
            'location_dest_id': cust_location.id,
        })
        move._action_confirm()
```
And not even bother creating the picking in the first place. As the move has a `picking_type_id` and no `picking_id`, it will be assigned to a picking (or create one if none is suitable). Which takes a lot less line to write. :slightly_smiling_face: